content/docs/cosidriver/features/objectscale.md

```diff
@@ -0,0 +1,327 @@
+
+```yaml
+apiVersion: objectstorage.k8s.io/v1alpha1
+kind: BucketAccess
+metadata:
+  name: my-bucketaccess
+  namespace: my-namespace
+spec:
+  bucketClaimName: my-bucketclaim
+  protocol: S3
+  bucketAccessClassName: my-bucketaccessclass
+  credentialsSecretName: my-s3-secret
+```
+
+#### `spec.protocol`
+
+> ⚠ **WARNING:** this field is case sensitive, and the provisioning will fail if protocol is not set exactly to *S3*.
+
+`spec.protocol` is the name of the Protocol that this access credential is supposed to support.
+
+#### Unsupported options
+
+- `spec.serviceAccountName=...` - is the name of the serviceAccount that COSI will map to the object storage provider service account when IAM styled authentication is specified. As the IAM style authentication is not supported, this field is also unsupported.
+- `spec.protocol=...` - Protocols are the set of data API this bucket is required to support. From protocols specified by COSI (`v1alpha1`), Dell ObjectScale platform only supports the `S3` protocol. Protocols `Azure` and `GCS` MUST NOT be used.
+
+## Provisioning Buckets
+
+### Kubernetes Administrator Steps
+
+The first and foremost step to do before you can start provisioning object storage, is to create a `BucketClass`. The `BucketClass` is an object that defines the provisioning and management characteristics of `Bucket` resources. It acts as an abstraction layer between users (such as applications or pods) and the underlying object storage infrastructure. `BucketClass` allows you to dynamically provision and manage `Buckets` in a consistent and automated manner.
+
+The provided code snippet demonstrates how to create a `BucketClass`.
+
+```sh
+cat <<EOF | kubectl create --filename -
+apiVersion: objectstorage.k8s.io/v1alpha1
+kind: BucketClass
+metadata:
+  name: my-bucketclass
+driverName: cosi.dellemc.com
+deletionPolicy: Delete
+parameters:
+  id: "my.objectscale"
+EOF
+```
+
+### End-user Steps
+
+#### Greenfield Provisioning
+
+_Greenfield Provisioning_ means creating a new bucket from scratch, without any existing data.
+
+The provided code snippet demonstrates how to create a `BucketClaim` for greenfield provisioning.
+
+```sh
+cat <<EOF | kubectl create --namespace=my-namespace --filename -
+apiVersion: objectstorage.k8s.io/v1alpha1
+kind: BucketClaim
+metadata:
+  name: my-bucketclaim
+spec:
+  bucketClassName: my-bucketclass
+  protocols: [ 'S3' ]
+EOF
+```
+
+> ℹ️ **NOTE:** remember to replace _my-namespace_, _my-bucketclass_ and _my-bucketclaim_ with actual values.
+
+#### Brownfield Provisioning
+
+_Brownfield Provisioning_ means using an existing bucket, that can already contain the data. This differs slightly from _Greenfield Provisioning_, as we need to create both `Bucket` and `BucketClaim` manually.
+
+The provided code snippet demonstrates how to create `Bucket` and `BucketClaim` for brownfield provisioning.
```

Comment:
The following example shows how to create `Bucket` and `BucketClaim` for brownfield provisioning.